src/applications/personalization/profile360/actions/index.js

```diff
@@ -11,12 +13,15 @@ export const FETCH_ADDRESS_CONSTANTS_SUCCESS =
 
 export function fetchHero() {
   return async dispatch => {
-    dispatch({
-      type: FETCH_HERO_SUCCESS,
-      hero: {
-        userFullName: await getData('/profile/full_name'),
-      },
-    });
+    dispatch({ type: FETCH_HERO });
+
+    await getData('/profile/full_name')
```

Comment:
Hmmm.... how did you test that these changes handle a server error properly? I'm not sure if you can chain `then` and `catch` calls on the response from `getData()`. I think you need to store the result of `getData()` and then, depending on what is in the response, either dispatch the success or failure.

Reply:
Updated! Thanks for the thorough review.